lib/spack/spack/spec.py

```diff
@@ -1954,6 +1959,44 @@ def from_json(stream):
             tty.debug(e)
             raise sjson.SpackJSONError("error parsing JSON spec:", str(e))
 
+    @staticmethod
+    def from_detection(spec_str, extra_attributes=None):
+        """Construct a spec from a spec string determined during external
+        detection and attach extra attributes to it.
+
+        Args:
+            spec_str (str): spec string
+            extra_attributes (dict): dictionary containing extra attributes
+
+        Returns:
+            Spec object (external spec)
```

Comment:
```suggestion
            spack.spec.Spec: external spec
```
I think this is the proper syntax for napoleon/sphinx